src/main.rs

```diff
@@ -32,91 +31,171 @@ use std::io::prelude::*;
+            .arg(Arg::with_name("message")
+                .short("m")
+                .long("message")
+                .value_name("COMMIT-MESSAGE")
+                .help("Commit message that will be used on import")
+                .default_value("cobalt site import")
+                .takes_value(true)))
+        .subcommand(SubCommand::with_name("serve")
+            .about("build and serve the cobalt project at the source dir")
+            .arg(Arg::with_name("port")
+                .short("P")
+                .long("port")
+                .value_name("INT")
+                .help("Port to serve from")
+                .default_value("3000")
+                .takes_value(true)))
+        .subcommand(SubCommand::with_name("watch")
+            .about("build, serve, and watch the project at the source dir")
+            .arg(Arg::with_name("port")
+                .short("P")
+                .long("port")
+                .value_name("INT")
+                .help("Port to serve from")
+                .default_value("3000")
+                .takes_value(true)))
+        .subcommand(SubCommand::with_name("import")
+            .about("moves the contents of the dest folder to the gh-pages branch")
+            .arg(Arg::with_name("branch")
+                .short("b")
+                .long("branch")
+                .value_name("BRANCH")
+                .help("Branch that will be used to import the site to")
+                .default_value("gh-pages")
+                .takes_value(true))
+            .arg(Arg::with_name("message")
+                .short("m")
+                .long("message")
+                .value_name("COMMIT-MESSAGE")
+                .help("Commit message that will be used on import")
+                .default_value("cobalt site import")
+                .takes_value(true)))
+        .get_matches();
+
+    let (command, matches) = match global_matches.subcommand() {
+        (command, Some(matches)) => (command, matches),
+        (_, None) => unreachable!(),
     };
 
-    if matches.opt_present("h") {
-        print_usage(opts);
-        return;
-    }
-
-    if matches.opt_present("version") {
-        println!("0.3.0");
-        return;
-    }
-
     let format = |record: &LogRecord| {
         let level = format!("[{}]", record.level()).to_lowercase();
         format!("{:8} {}", level, record.args())
     };
 
     let mut builder = LogBuilder::new();
     builder.format(format);
-    builder.filter(None, LogLevelFilter::Info);
-
-    if matches.opt_present("debug") {
-        builder.filter(None, LogLevelFilter::Debug);
-    }
-
-    if matches.opt_present("trace") {
-        builder.filter(None, LogLevelFilter::Trace);
-    }
 
-    if matches.opt_present("silent") {
-        builder.filter(None, LogLevelFilter::Off);
+    match matches.value_of("log-level").or(global_matches.value_of("log-level")) {
+        Some("error") => {
+            builder.filter(None, LogLevelFilter::Error);
+        }
```

Comment:
Nit: can you leave out the braces in this match block? Those are a bit cluttering for tons of single-line statements. (Unless this is because of rustfmt of course)